VoxelSniperCore/src/main/java/com/github/kevindagame/util/schematic/SchematicReader.kt

```diff
@@ -0,0 +1,123 @@
+package com.github.kevindagame.util.schematic
+
+import com.github.kevindagame.VoxelSniper
+import net.sandrohc.schematic4j.SchematicUtil
+import net.sandrohc.schematic4j.schematic.types.SchematicPosInteger
+import java.io.File
+
+object SchematicReader {
+
+    private fun readSchematic(file: File): VoxelSchematic {
+        val schematic = SchematicUtil.load(file)
+        val voxelSchematicBuilder = VoxelSchematicBuilder()
+
+        voxelSchematicBuilder.name = file.nameWithoutExtension
+
+        for (y in 0 until schematic.height) {
+            for (x in 0 until schematic.width) {
+                for (z in 0 until schematic.length) {
+                    val pos = SchematicPosInteger(x, y, z)
+                    val block = schematic.getBlock(pos)
+                    voxelSchematicBuilder.addBlock(x.toDouble(), y.toDouble(), z.toDouble(), block)
+                }
+            }
+        }
+        return voxelSchematicBuilder.build()
+    }
+
+    private fun readSchematics(folder: File): MutableList<VoxelSchematic> {
+        if (!folder.exists()) {
+            throw IllegalArgumentException("Folder $folder does not exist")
+        }
+        val schematicFiles = folder.listFiles()
+        if (schematicFiles != null) {
+            if(schematicFiles.none { it.extension == "schem" }) {
+                throw IllegalArgumentException("Folder $folder does not contain any schematics")
+            }
+        }
+        val schematics = mutableListOf<VoxelSchematic>()
+        if (schematicFiles != null) {
+            for (schematic in schematicFiles) {
+                if (schematic.isFile) {
+                    if (schematic.extension != "schem") {
+                        continue
+                    }
+                    schematics.add(readSchematic(schematic))
```

Comment:
```suggestion
                    if (schematic.extension == "schem") {
                        schematics.add(readSchematic(schematic))
                    }
```
I think I like this just a little better